astropy/io/ascii/core.py

```diff
@@ -618,6 +618,48 @@ def colnames(self):
         return tuple(col.name if isinstance(col, Column) else col.info.name
                      for col in self.cols)
 
+    def remove_columns(self, names):
+        """
+        Remove several columns from the table.
+
+        Parameters
+        ----------
+        names : list
+            A list containing the names of the columns to remove
+        """
+        colnames = self.colnames
+        for name in names:
```

Comment:
Maybe just
```
if any(name not in colnames for name in names):
   raise(...)
```

Reply:
In principle, with the new behaviour
`converters={'ntp_seconds': [convert_numpy(np.int64)]}` 
becomes the correct usage.
But in a way this is complementary to my comment in #9701 on pathological `converters` with conflicting values - meaning one could consider as a convenience and backwards-compatibility option to consider any converters passed by default, automatic _or_ user names, as in
```
converters.set_default('A', converters['a'])
converters.set_default('A', converters['col1'])
```
so that additional tests
```
    converters = {
        'a': [ascii.convert_numpy(float)],
        'b': [ascii.convert_numpy(int)],
        'c': [ascii.convert_numpy(str)]
    }
    t = ascii.read(csv_text, format='csv', names=names, converters=converters)
    assert t['A'].dtype.kind == 'f'
    assert t['B'].dtype.kind == 'i'
    assert t['C'].dtype.kind == 'U'

    converters = {
        'col1': [ascii.convert_numpy(float)],
        'col2': [ascii.convert_numpy(int)],
        'col3': [ascii.convert_numpy(str)]
    }
    t = ascii.read(csv_text, format='csv', names=names, converters=converters)
    assert t['A'].dtype.kind == 'f'
    assert t['B'].dtype.kind == 'i'
    assert t['C'].dtype.kind == 'U'
```
would pass.
Unfortunately I don't see any obvious place to implement this where all the different colnames would be readily available.

Reply:
Yes, good to have the name (walrus to the rescue eventually?).

And agreed with choosing one over the other, even if it is somewhat painful.

Reply:
Agreed, too. The user after all selected the custom names, so it's reasonable to expect them to use those names for the converters, too.